app/models/github_blob.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "safe_yaml/load"
+
+class GitHubBlob
+  # From jekyll/jekyll
+  # https://github.com/jekyll/jekyll/blob/74373baa550282a8630368e7b609ca9370f6d560/lib/jekyll/document.rb#L13
+  YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
+
+  attr_reader :data, :body
+
+  def initialize(github_repository, sha, **options)
+    @blob = github_repository.client.blob(github_repository.full_name, sha, options)
+    read_contents
+  end
+
+  # Public: Get the utf-8 formated content of the blob
+  #
+  # Returns a string.
+  def utf_content
```

Comment:
This is 1-1 the same as `decoded_content`, we can probably just take the body from decoded content and throw it in here.